source/contracts/extensions/MarketFeeCalculator.sol

```diff
@@ -3,45 +3,76 @@ pragma solidity ^0.4.13;
-        uint256 _previousTimestamp = _reportingWindow.getStartTime() - 1;
-        IUniverse _universe = _reportingWindow.getUniverse();
-        IReportingWindow _previousReportingWindow = _universe.getReportingWindowByTimestamp(_previousTimestamp);
+        IReportingWindow _previousReportingWindow = _reportingWindow.getPreviousReportingWindow();
+        uint256 _totalMarketsInPreviousWindow = _reportingWindow.getNumMarkets();
+        uint256 _indeterminateMarketsInPreviousWindow = _reportingWindow.getNumIndeterminateMarkets();
         uint256 _previousValidityBondInAttoeth = validityBondInAttoeth[_previousReportingWindow];
-        if (_previousValidityBondInAttoeth == 0) {
-            _previousValidityBondInAttoeth = 1 * 10 ** 16;
-        }
-        uint256 _targetIndeterminateMarketsPerHundred = 1;
-        _currentValidityBondInAttoeth = _previousValidityBondInAttoeth * _targetIndeterminateMarketsPerHundred * _totalMarketsInPreviousWindow / _indeterminateMarketsInPreviousWindow / 100;
+
+        _currentValidityBondInAttoeth = calculateValidityBond(_indeterminateMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INDETERMINATE_MARKETS_DIVISOR, _previousValidityBondInAttoeth);
         validityBondInAttoeth[_reportingWindow] = _currentValidityBondInAttoeth;
         return _currentValidityBondInAttoeth;
     }
 
-    function getTargetReporterGasCosts() constant public returns (uint256) {
-        // TODO: get number of registration tokens issued last period
-        // TODO: get target reporter count + wiggle room
-        // TODO: calculate estimated reporters per market
-        uint256 _estimatedReportsPerMarket = 10;
-        // TODO: figure out what the number actually is
-        uint256 _gasToReport = 100000;
-        // we double it to ensure we have more than enough rather than not enough
-        uint256 _estimatedReportingGas = _gasToReport * _estimatedReportsPerMarket * 2;
-        // TODO: multiply this by average gas costs of reporters historically
-        return _estimatedReportingGas;
+    function calculateValidityBond(uint256 _indeterminateMarkets, uint256 _totalMarkets, uint256 _targetIndeterminateMarketsDivisor, uint256 _previousValidityBondInAttoeth) constant public returns (uint256) {
+        if (_totalMarkets == 0) {
+            return DEFAULT_VALIDITY_BOND;
+        }
+        if (_previousValidityBondInAttoeth == 0) {
+            _previousValidityBondInAttoeth = DEFAULT_VALIDITY_BOND;
+        }
+        
+        uint256 _targetIndeterminateMarkets = _totalMarkets.div(_targetIndeterminateMarketsDivisor);
+
+        if (_indeterminateMarkets <= _targetIndeterminateMarkets) {
+            return _indeterminateMarkets
+                .mul(_previousValidityBondInAttoeth)
+                .mul(_targetIndeterminateMarketsDivisor)
+                .div(_totalMarkets)
+                .div(2)
+            .add(_previousValidityBondInAttoeth.div(2));
```

Comment:
Indentation on this line is off.  If you did this because Solium complained, move the `;` to the next line.  I have a comment somewhere else with a link to a GitHub issue as a reminder that we should put the `;` back on the previous line once the Solium issue is resolved.